File: backend/GraficheVeneziane/src/main/java/graficheveneziane/graficheveneziane/model/Fornitore.java
package graficheveneziane.graficheveneziane.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@Table(name = "fornitori")
@AllArgsConstructor
@NoArgsConstructor
public class Fornitore {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_fornitore;

    @Column
    private String nome_fornitore;
    @Column
    private String email;

    @Column
    private String referente;

    @Column
    private String numero_cellulare;

    @Column
    private String numero_fisso;

    @Column
    private String via;

    @Column
    private String cap;

    @Column
    private String provincia;

    @Column
    private String p_iva;

}

File: fe/graficheFE/src/app/model/Fornitore.ts
export class Fornitore {
    id_fornitore: number | undefined;
    nome_fornitore: string;
    email: string;
    referente: string;
    numero_cellulare: string;
    numero_fisso: string;
    via: string;
    cap: string;
    provincia: string;
    p_iva: string;
  
    constructor(
      id_fornitore?: number,
      nome_fornitore: string = '',
      email: string = '',
      referente: string = '',
      numero_cellulare: string = '',
      numero_fisso: string = '',
      via: string = '',
      cap: string = '',
      provincia: string = '',
      p_iva: string = ''
    ) {
      this.id_fornitore = id_fornitore;
      this.nome_fornitore = nome_fornitore;
      this.email = email;
      this.referente = referente;
      this.numero_cellulare = numero_cellulare;
      this.numero_fisso = numero_fisso;
      this.via = via;
      this.cap = cap;
      this.provincia = provincia;
      this.p_iva = p_iva;
    }
  }
  